apps/dh/api-dh/source/DataHub.WebApi/Startup.cs

```diff
@@ -141,11 +146,38 @@ public void Configure(IApplicationBuilder app, IWebHostEnvironment environment)
             app.UseEndpoints(endpoints =>
             {
                 endpoints.MapControllers();
-                endpoints.MapHealthChecks("/health").AllowAnonymous();
+
+                // Health check
+                endpoints.MapLiveHealthChecks();
+                endpoints.MapReadyHealthChecks();
             });
         }
 
-        private void AddDomainClients(IServiceCollection services)
+        protected virtual void SetupHealthEndpoints(IServiceCollection services, ApiClientSettings apiClientSettingsService)
+        {
+            const string liveEndpointPath = "/monitor/live";
+            var chargesLiveHealthUrl = apiClientSettingsService.ChargesBaseUrl == string.Empty
+                ? throw new ArgumentException()
+                : new Uri(apiClientSettingsService.ChargesBaseUrl + liveEndpointPath);
+            var marketParticipantLiveHealthUrl = apiClientSettingsService.MarketParticipantBaseUrl == string.Empty
+                ? throw new ArgumentException()
+                : new Uri(apiClientSettingsService.MarketParticipantBaseUrl + liveEndpointPath);
+            var messageArchiveLiveHealthUrl = apiClientSettingsService.MessageArchiveBaseUrl == string.Empty
+                ? throw new ArgumentException()
+                : new Uri(apiClientSettingsService.MessageArchiveBaseUrl + liveEndpointPath);
+            var wholesaleLiveHealthUrl = apiClientSettingsService.WholesaleBaseUrl == string.Empty
+                ? throw new ArgumentException()
+                : new Uri(apiClientSettingsService.WholesaleBaseUrl + liveEndpointPath);
+
+            services.AddHealthChecks()
+                .AddLiveCheck()
+                .AddServiceHealthCheck("charges", chargesLiveHealthUrl)
+                .AddServiceHealthCheck("marketParticipant", marketParticipantLiveHealthUrl)
+                .AddServiceHealthCheck("messageArchive", messageArchiveLiveHealthUrl)
+                .AddServiceHealthCheck("wholesale", wholesaleLiveHealthUrl);
+        }
+
+        private ApiClientSettings AddDomainClients(IServiceCollection services)
         {
             var apiClientSettings = Configuration.GetSection("ApiClientSettings").Get<ApiClientSettings>();
```

Comment:
Move this to `ConfigureServices` and add a parameter to `AddDomainClients`. Then consider if `AddDomainClients` and `SetupHealthEndpoints` can be called just after one another, so we easily see their relationship. But beware, sometimes the order of registration is important.

Reply:
done